lib/Test/PostgreSQL.pm

```diff
@@ -389,16 +425,26 @@ method start() {
 # It could probably be made more sane.
 method _find_port_and_launch() {
   my $tries = 10;
-  my $port = $self->base_port;
+
   srand(); # Re-seed the RNG in case the caller forked the process
+
+  # There is a significant chance that there might be more than
+  # one Test::Postgresql test in flight so we better randomize
+  # the port before even trying to start the first time. Does
+  # no harm if there is no concurrent Postgres running; if there is
+  # then our process will start up a bit faster.
+  my $port = $self->base_port + int(rand(10)) + 1;
+
   # try by incrementing port number until PostgreSQL starts
   while (1) {
-    my $good = try {
+    my $good;
+    try {
+      #warn "Trying to start postgres on port $port...";
```

Comment:
Rather than comment out the log messages, maybe set them to only print if a debug flag is set?

Reply:
I agree here, although I don't want to see these warnings by default, a debug flag which causes them to be printed to STDOUT would be a nice improvement.

Reply:
There are pros and cons to this... It's been some time since I worked on this and if I remember correctly, I did consider adding a flag and decided not to, because these warnings are needed _very_ rarely and only by us package developers while troubleshooting; it shouldn't be much of a burden to uncomment a line of code you're working with.

On the other hand, a flag would need to be checked at run time and that's a cost that every user out there will have to pay constantly with very low probability of ever needing to turn it on. A small cost for sure but these things can add up to seconds in large test runs.